pkg/discovery/processor/cluster_processor.go

```diff
@@ -6,36 +6,116 @@ import (
 	"github.com/turbonomic/kubeturbo/pkg/cluster"
 	"github.com/turbonomic/kubeturbo/pkg/discovery/metrics"
 	"github.com/turbonomic/kubeturbo/pkg/discovery/repository"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/util"
+	"github.com/turbonomic/kubeturbo/pkg/kubeclient"
+	"math/rand"
+	"strings"
 )
 
 // Class to query the cluster data from the Kubernetes API server and create the KubeCluster entity
 // to represent the cluster, the nodes and the namespaces.
 type ClusterProcessor struct {
 	ClusterInfoScraper *cluster.ClusterScraper
+	NodeScrapper       *kubeclient.KubeletClient
 }
 
 // Query the Kubernetes API Server to get the cluster nodes and namespaces.
-// Creates a KubeCluster entity to represent the cluster and its entities and resources
-func (processor *ClusterProcessor) ProcessCluster() (*repository.KubeCluster, error) {
+func (processor *ClusterProcessor) ConnectCluster(checkAllNodes bool) (*repository.KubeCluster, error) {
 	svcID, err := processor.ClusterInfoScraper.GetKubernetesServiceID()
 	if err != nil {
 		return nil, fmt.Errorf("Cannot obtain service ID for cluster %s\n", err)
 	}
+	kubeCluster := repository.NewKubeCluster(svcID)
+
+	// Nodes
+	err = processor.ConnectToNodes(checkAllNodes)
+
+	return kubeCluster, err
+}
 
-	kubeCluster := &repository.KubeCluster{
-		Name:       svcID,
-		Nodes:      make(map[string]*repository.KubeNode),
-		Namespaces: make(map[string]*repository.KubeNamespace),
+func (processor *ClusterProcessor) ConnectToNodes(checkAllNodes bool) error {
+	nodeList, err := processor.ClusterInfoScraper.GetAllNodes()
+	if err != nil {
+		return fmt.Errorf("Error getting nodes for cluster : %s\n", err)
 	}
+	glog.V(2).Infof("There are %d nodes\n", len(nodeList))
 
-	// Nodes
-	nodes, err := processor.processNodes(kubeCluster.Name)
+	var connectionErrors []error
+	var connectionMsgs []string
+
+	n := rand.Int() % len(nodeList)
```

Comment:
Let's add some comments for the logic.

From my understanding, if _checkAllNodes_ is false, we only check one random node. If the random node is not responding, the validation will fail. Correct? If so, it may result in some inconsistent validation result. E.g., let's say there are two nodes with one node healthy and the other not responsive. If we click validation button 10 times, about 5 times will fail and others succeed. Is this behavior we want?